### PR TITLE
🤘 Implementation for #59 to support username and password as optional

### LINKED ISSUE #59: Make username and password optional for Self-Hosted runner use cases
## Context
Related issue: https://github.com/dawidd6/action-send-mail/issues/10

@dawidd6 Thanks for providing this really handy action.

## Problem
I am relating the issue #10 where this was addressed, and I can see a use case where having username/password optional can be really handy. This is for self-hosted runners, where they have access to internal infrastructure and smtp services. In those cases, the smtp server can work without authentication and it's secure from the access through the internet. If this is or it's not a questionable practice within the enterprise environment is something that can be discussed, but certainly there is audience where this may be interesting.

## Solution
I would suggest, and if you agree I am happy to create a PR, that we make username/password optional and put a warning message in the case people do it to make sure they are aware of the risk.

Optionally we can set a boolean parameter to ack that they are doing something that may potentially be insecure if you want to make this more explicit.

## Proposed changes
The changes required would be:
- Remove the required restriction for username/password
- Add a warning message printed on the log
- (Optional) add an ack parameter to explicitly say you want to be using a non authenticated smtp request
- Add documentation to the readme

What are your thoughts? 😸

### PR DESCRIPTION
**Pull request by droidpl**
This PR fixes #59 

Adding support for optional fields `username` and `password`

⚠️ Draft PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,3 +63,8 @@ Changes in Gmail settings may be necessary to get this action to work.
 3. If the Gmail account you're trying to use in this Action is already 2FA (Two Factor Authentication) enabled, the 2FA password will need to be provided as well, which isn't included in the default template.
 
 Users who have had problems have reported success by doing each of these three steps or by doing the first two steps and using a Gmail account that didn't have 2FA enabled.
+
+### Unauthenticated login (username/password fields)
+
+The parameters `username` and `password` are set as optional to support self-hosted runners access to on-premise infrastructure. If
+you are accessing public email provided make sure you provide a username/password authentication throuth [GitHub Secrets storage](https://docs.github.com/en/actions/reference/encrypted-secrets) to make the email delivery secure.
```
Comment (dawidd6) [inline README.md:70]:
```suggestion
you are accessing public email provided make sure you provide a username/password authentication through [GitHub Secrets storage](https://docs.github.com/en/actions/reference/encrypted-secrets) to make the email delivery secure.
```

File: main.js [hunk 2]
```diff
@@ -48,17 +48,25 @@ async function main() {
...
             secure: secure ? true : serverPort == "465",
-            auth: {
-                user: username,
-                pass: password,
-            },
             tls: ignoreCert ? {
                 rejectUnauthorized: false
-            } : undefined
+            } : undefined,
+            auth,
```
Comment (dawidd6) [inline main.js:69]:
```suggestion
            auth: username && password ? {
                user: username,
                pass: password,
            } : undefined,
```

Comment (dawidd6) [reply main.js:69]: Just like with ignoreCert.

Comment (droidpl) [reply main.js:69]: I'll submit an additional commit with a change here 👍

**Comment by droidpl** [comment]
Fixed your comments. Tomorrow I will run a quick test and let you know 👍

File: README.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ Some features:
 
 ## Usage
 
+### Example
```
Comment (dawidd6) [inline README.md:15]:
```suggestion
```

**Comment by droidpl** [comment]
I've just tested this @dawidd6 and seems to work fine. Also submitted a fix for the `ignoreCert` and `secure` properties. It seems that when passing "false" it is taking it as string thus the `?` resolves to true, setting it as true even if someone passes false.

**Comment by dawidd6** [comment]
Instead of `===` we can use `==` I guess, as I always did in my Actions.

Also please revert all those changes regarding logging. I mean `core.info(JSON.stringify...` and around. There were complaints about printing out some sensitive info in the past.

Then I think it will be good to go.

**Comment by droidpl** [comment]
I will revert the log changes. On the equality, I believe it's better to do `===` as that will compare both the type and the value in Javascript. [Here](https://www.jstips.co/en/javascript/use_===_instead_of_==/#:~:text=)%20operator%20performs%20an%20automatic%20type,faster%20(jsPref)%20than%20%3D%3D%20.) you can see some examples of things that are weird with the `==` so I suggest we leave that as is to avoid those weird scenarios of javascript

**Comment by droidpl** [comment]
Should be now ready for review

File: test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const nodemailer = require("nodemailer")
```
Comment (dawidd6) [inline test.js:1]: I believe this file should not be here

Comment (droidpl) [reply test.js:1]: jish, let me force push this out of the history 🤦

**Comment by droidpl** [comment]
Should be all fixed now. Apologies for the comes and goes

**Comment by dawidd6** [comment]
Thanks. Sure I'll tag one.

**Comment by droidpl** [comment]
Thanks

**Pull request merged**